mmv1/third_party/terraform/tests/resource_network_services_endpoint_policy_test.go.erb

```diff
@@ -0,0 +1,92 @@
+<% autogen_exception -%>
+package google
+<% unless version == 'ga' -%>
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+)
+
+func TestAccNetworkServicesEndpointPolicy_update(t *testing.T) {
+	t.Parallel()
+
+	endpointPolicyName := fmt.Sprintf("tf-test-endpoint-policy-%s", RandString(t, 10))
+
+	VcrTest(t, resource.TestCase{
+		PreCheck:     func() { AccTestPreCheck(t) },
+		ProtoV5ProviderFactories: ProtoV5ProviderFactories(t),
+		CheckDestroy: testAccCheckNetworkServicesEndpointPolicyDestroyProducer(t),
+		Steps: []resource.TestStep{
+			{
+				Config: testAccNetworkServicesEndpointPolicy_basic(endpointPolicyName),
+			},
+			{
+				ResourceName:      "google_network_services_endpoint_policy.foobar",
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+			{
+				Config: testAccNetworkServicesEndpointPolicy_update(endpointPolicyName),
+			},
+			{
+				ResourceName:      "google_network_services_endpoint_policy.foobar",
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+		},
+	})
+}
+
+func testAccNetworkServicesEndpointPolicy_basic(endpointPolicyName string) string {
+	return fmt.Sprintf(`
+resource "google_network_services_endpoint_policy" "foobar" {
+  name                   = "%s"
+  labels                 = {
+    foo = "bar"
+  }
+  description            = "my description"
+  type                   = "SIDECAR_PROXY"
+  traffic_port_selector {
+    ports = ["8081"]
+  }
+  endpoint_matcher {
+    metadata_label_matcher {
+      metadata_label_match_criteria = "MATCH_ANY"
+      metadata_labels {
+        label_name = "foo"
+        label_value = "bar"
+      }
+    }
+  }
+}
+`, endpointPolicyName)
+}
+
+func testAccNetworkServicesEndpointPolicy_update(endpointPolicyName string) string {
+	return fmt.Sprintf(`
+resource "google_network_services_endpoint_policy" "foobar" {
+  name        = "%s"
+  labels                 = {
+    foo = "bar"
+  }
+  description            = "update description"
+  type                   = "GRPC_SERVER"
+  traffic_port_selector {
+    ports = ["8081"]
```

Comment:
```suggestion
    ports = []
```

Let's make sure that updating to empty works here.

Reply:
Tested by removing the traffic_port_selector section. Otherwise, the test fails although the update is successful. This is because the API returns a response without the traffic_port_selector section.

Reply:
Cool! That's fine.